implementation/src/main/java/io/smallrye/config/SmallRyeConfigBuilder.java

```diff
@@ -184,23 +196,27 @@ public SmallRyeConfigBuilder withSources(Collection<ConfigSource> configSources)
     }
 
     public SmallRyeConfigBuilder withSources(ConfigSourceFactory... configSourceFactories) {
-        Stream.of(configSourceFactories).forEach(configSourceFactory -> {
+        for (ConfigSourceFactory configSourceFactory : configSourceFactories) {
             sources.add(new ConfigurableConfigSource(configSourceFactory));
-        });
+        }
         return this;
     }
 
     public SmallRyeConfigBuilder withInterceptors(ConfigSourceInterceptor... interceptors) {
-        this.interceptors.addAll(Stream.of(interceptors)
-                .map(InterceptorWithPriority::new)
-                .collect(Collectors.toList()));
+        List<InterceptorWithPriority> interceptorWithPriorities = new ArrayList<>();
+        for (ConfigSourceInterceptor interceptor : interceptors) {
+            interceptorWithPriorities.add(new InterceptorWithPriority(interceptor));
+        }
+        this.interceptors.addAll(interceptorWithPriorities);
```

Comment:
Is there a reason why you introduce an intermediary list here?

Reply:
Yes, no point.